crates/sozo/src/init.rs

```diff
@@ -1,30 +1,17 @@
 use std::env::current_dir;
-use std::path::{Path, PathBuf};
-use std::{fs, io};
+use std::error::Error;
+use std::path::PathBuf;
 
 use clap::Args;
+use git2::Repository;
 
 #[derive(Args, Debug)]
 pub struct InitArgs {
     #[clap(help = "Target directory")]
     path: Option<PathBuf>,
 }
 
-fn copy_dir_all(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> io::Result<()> {
-    fs::create_dir_all(&dst)?;
-    for entry in fs::read_dir(src)? {
-        let entry = entry?;
-        let ty = entry.file_type()?;
-        if ty.is_dir() {
-            copy_dir_all(entry.path(), dst.as_ref().join(entry.file_name()))?;
-        } else {
-            fs::copy(entry.path(), dst.as_ref().join(entry.file_name()))?;
-        }
-    }
-    Ok(())
-}
-
-pub fn run(args: InitArgs) {
+pub fn run(args: InitArgs) -> Result<(), Box<dyn Error>> {
```

Comment:
I think we should follow the foundry approach and use the users local git installation, rather than bundling git itself.

https://github.com/foundry-rs/foundry/blob/a5c27411c7e5d1dd574df758060da8ad501ecbde/cli/src/cmd/forge/init.rs#L62